pkg/controller/cloud/node_controller_test.go

```diff
@@ -407,84 +340,55 @@ func TestZoneInitialized(t *testing.T) {
 					},
 				},
 				Spec: v1.NodeSpec{
-					Taints: []v1.Taint{
-						{
-							Key:    cloudproviderapi.TaintExternalCloudProvider,
-							Value:  "true",
-							Effect: v1.TaintEffectNoSchedule,
-						},
-					},
+					ProviderID: "aws://12345",
```

Comment:
At some point it would be nice to get a properly functioning sample cloud provider, with its own ID and use that. No chance of bleed over with/from an existing provider.

Reply:
I learned during this refactor that there's a few issues with `FakeCloud` and as a result we are not testing certain code paths (like most of the *ByProviderID methods), I intend to clean this up in a future PR.

Reply:
Sounds good. I assume we have a tracking bug?

Reply:
Not yet. I don't understand the full scope of the issues yet, but I'll create one when I do